### PR TITLE
bdev: fix FTBFS on FreeBSD, keep the huge paged read buffers.

### PR DESCRIPTION
**Pull request by rzarzynski**
Special thanks to Willem Jan Withagen!

Signed-off-by: Radoslaw Zarzynski <rzarzyns@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by wjwithagen** [comment]
Lets see if my master builder is happy as well.
Should be, but beter dafe.

**Comment by wjwithagen** [comment]
It would deserve some better error handling for FreeBSD as well.
But this is fine enough for now.

**Review by ifed01 — APPROVED** [review]
Personally I'd leave the mmap call out of ifdef block and wrap an assignment to map_mode var with it only:
int map_mode;
```c++
#ifdef (free_bsd)
map_mode = ...
#else
map_mode = ...
#endif
::mmap(...,map_mode,...);
```
but not a big deal...

**Comment by wjwithagen** [comment]
Ping

**Comment by rzarzynski** [comment]
@ifed01: Thanks for pointing this out! Minimized the patch by guarding the flags directly.

**Comment by rzarzynski** [comment]
jenkins retest this please

```
4 tests failed out of 275

Total Test time (real) = 3600.44 sec

The following tests FAILED:
	 36 - smoke.sh (Timeout)
	175 - mgr-dashboard-smoke.sh (Failed)
	179 - unittest_rocksdb_option (Failed)
	190 - safe-to-destroy.sh (Timeout)
```

**Review by aclamk — APPROVED** [review]

File: src/blk/kernel/KernelDevice.cc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/file.h>
+#include <sys/mman.h>
```
Comment (rzarzynski) [inline src/blk/kernel/KernelDevice.cc:22]: Made the include unconditional per Adam's note.

**Comment by rzarzynski** [comment]
jenkins test api

**Pull request merged**